ncolony/ctllib.py

```diff
@@ -53,23 +53,31 @@ def _dumps(stuff):
 
 
 # pylint: disable=too-many-arguments,too-many-locals
-def add(places, name, cmd, args, env=None, uid=None, gid=None, extras=None,
-        env_inherit=None):
+def add(places, name, cmd, args,
+        env=None, uid=None, gid=None, extras=None,
+        env_inherit=None, _cmd_and_args=None):
```

Comment:
`_cmd_and_args` is effectively a public API change, but it should be backwards compatible. If you feel strongly about it, I can rename this function to `_add` and introduce a new `add` wrapper without the new argument.